lib/bom.gradle

```diff
@@ -14,14 +14,48 @@ configure(projectsWithFlags('bom')) {
 
     dependencies {
         constraints {
+            boolean isNeededCheck = false
+            def bomGroups, subs
+            if (rootProject.ext.has('bomGroups')) {
+                bomGroups = rootProject.ext.bomGroups
+                if (!(bomGroups instanceof Map)) {
+                    throw new IllegalStateException("bomGroups must be a Map: ${bomGroups}")
+                }
+                if (!bomGroups.containsKey(project.path)) {
+                    logger.warn(project.name + " won't be included to bomGroups property")
+                    isNeededCheck = true
+                }
+                subs = bomGroups.get(project.path)
+                if (!(subs.value instanceof List)) {
+                    throw new IllegalStateException("bomGroups' value must be a List: ${subs.value}")
+                }
+            }
+
             projectsWithFlags('java', 'publish').toList().sort { a, b ->
                 def groupComparison = "${a.group}".compareTo("${b.group}")
                 if (groupComparison != 0) {
                     return groupComparison
                 }
                 return "${a.ext.artifactId}".compareTo("${b.ext.artifactId}")
             }.each { p ->
-                api "${p.group}:${p.ext.artifactId}:${p.version}"
+                if (rootProject.ext.has('bomGroups')) {
+                    if (!bomGroups.containsKey(project.path)) {
+                        api "${p.group}:${p.ext.artifactId}:${p.version}"
+                    } else {
+                        if (subs.contains(p.path)) {
+                            api "${p.group}:${p.ext.artifactId}:${p.version}"
+                        } else if (p.path.startsWith(project.path)) {
+                            logger.warn(p.name + " won't be included in " + project.name + " bom")
+                            isNeededCheck = true
+                        }
+                    }
+                } else {
+                    api "${p.group}:${p.ext.artifactId}:${p.version}"
+                }
+            }
+
+            if (isNeededCheck) {
+                logger.warn("If not intended, Please check bomGroups property")
```

Comment:
I think each log message has its own context. But this log message seems to depend on the preview context. Some users may don't get what is intended from `If not intended`. How about moving that message when `isNeededCheck` is set to true?
``java
logger.warn( "{} won't be included in {} bom. If not intended, Please check bomGroups property", p.name, project.name)
```

Reply:
Thanks! Applied your comment. Please check one more.